lib/rules/disallow-function-prototype-extension.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+var includes = require("../utils/includes");
+var extensionFixes = {
+  'property': "Use the computed('dependentKey', function() {}) syntax instead.",
+  'observes': "Use the observer('dependentKey', function() {}) syntax instead.",
+  'observesBefore': "Also, using before observers is deprecated."
+}
+var prohibitedFunctionMethods = Object.keys(extensionFixes);
+
+module.exports = function(context) {
+  return {
+    'CallExpression': function(node) {
+      if (
+        node.callee.object && (
+          node.callee.object.type === 'FunctionExpression' ||
+          node.callee.object.type === 'CallExpression'
+        ) &&
+        includes(prohibitedFunctionMethods, node.callee.property.name)) {
+
+        context.report({
+          node: node,
+          message: message(node)
+        });
+      }
+    }
+  };
+};
+
+function message(node) {
+  var propertyName = node.callee.property.name;
+  var message = "Avoid using function prototype extensions like '" + propertyName + "'. " +
```

Comment:
Again for consistency, are these messages descriptive or proscriptive? The other message is: ["Property on the namespace was accessed directly"](https://github.com/201-created/eslint-plugin-ember-rules/blob/master/lib/rules/accessing-ember-namespace.js#L12) which is descriptive, this message "Avoid using function prototype extensions" is proscriptive.